xdsl/dialects/arith.py

```diff
@@ -134,16 +133,15 @@ def __hash__(self) -> int:
 @irdl_op_definition
 class Addi(BinaryOperation):
     name: str = "arith.addi"
+
     lhs: Annotated[Operand, signlessIntegerLike]
     rhs: Annotated[Operand, signlessIntegerLike]
     result: Annotated[OpResult, signlessIntegerLike]
 
-    @staticmethod
-    def get(operand1: Union[Operation, SSAValue],
-            operand2: Union[Operation, SSAValue]) -> Addi:
+    def __init__(self, operand1: Union[Operation, SSAValue],
```

Comment:
Oh also, can you add a `result` field where one can add an arbitrary result type?
This should help if we are in the middle of a transformation, and while operands have the wrong types, we want the result to have the right one.